### PR TITLE
allow partials to still work when error level is ignore

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -226,6 +225,22 @@ def raise_error(self, message, token=None):
         if self.error_level == ErrorLevel.WARN:
             logging.error(self.error)
 
+    def expression(self, exp_class, **kwargs):
```
Comment (izeigerman) [inline sqlglot/parser.py:228]: Ah, nice. This is something I was thinking about myself 👍

File: sqlglot/parser.py [hunk 2]
```diff
@@ -226,6 +225,22 @@ def raise_error(self, message, token=None):
...
+        instance = exp_class(**kwargs)
+        self.validate(instance)
+        return instance
+
+    def validate(self, expression):
```
Comment (izeigerman) [inline sqlglot/parser.py:233]: [Nitpick/Optional] I would be more specific about what's being validated here and call it explicitly `validate_expression`.

File: sqlglot/parser.py [hunk 3]
```diff
@@ -226,6 +225,22 @@ def raise_error(self, message, token=None):
...
+        if self.error_level == ErrorLevel.IGNORE:
+            return
+
+        for k in expression.args:
+            if k not in expression.arg_types:
+                self.raise_error(f"Unexpected keyword: '{k}' for {expression.token_type}")
+        for k, v in expression.arg_types.items():
+            if v and expression.args.get(k) is None:
```
Comment (izeigerman) [inline sqlglot/parser.py:241]: This is something I realized after I merged my PR: should we also check for empty list here and not just None?

**Comment by izeigerman** [comment]
I think you missed a wrapper for one expression creation  here - https://github.com/tobymao/sqlglot/blob/a29c4d6f7c7a4633e96008cdb06ebe597cbd2660/sqlglot/parser.py#L990

File: sqlglot/parser.py (hunk 2 above)
Comment (tobymao) [reply sqlglot/parser.py:233]: done

File: sqlglot/tokens.py [hunk 4]
```diff
@@ -168,6 +168,10 @@ def number(cls, number):
     def string(cls, string):
         return cls(TokenType.STRING, string)
 
+    @classmethod
+    def identifier(cls, identifier):
```
Comment (izeigerman) [inline sqlglot/tokens.py:172]: Is this being used anywhere?

Comment (tobymao) [reply sqlglot/tokens.py:172]: not internally no, these methods are used externally

File: sqlglot/parser.py (hunk 3 above)
Comment (tobymao) [reply sqlglot/parser.py:241]: not sure about this one, need to double check, it breaks some stuff

Comment (tobymao) [reply sqlglot/parser.py:241]: hm i think it might work

**Pull request merged**